src/interceptors/response.ts

```diff
@@ -95,35 +63,87 @@ export class CacheResponseInterceptor<R, D>
 
       // Cache should not be used
       if (expirationTime === false) {
-        await this.rejectResponse(key);
+        await this.rejectResponse(response.id);
         return response;
       }
 
       ttl = expirationTime ? expirationTime : ttl;
     }
 
+    const data =
+      response.status == 304 && cache.data
+        ? (() => {
+            // Rust syntax <3
+            response.cached = true;
+            response.data = cache.data.data;
+            response.headers = cache.data.headers;
```

Comment:
new response may have sent new headers, you need to use the new ones, but fallback to the cached ones, at least for 'cache-control', 'etag', 'last-modified' headers

Reply:
Just `response.headers = { ...cache.data.headers, ...response.headers };` is ok?

Reply:
Seeing what we use the headers for, I think it does the job :)